app/models/speaker.rb

```diff
@@ -43,7 +43,10 @@ def format_links
   end
 
   def url_exists?
-    links.each_value { |url| errors.add(:links, 'This url is not valid') unless url_valid?(url) }
+    links.each do |key, raw_url|
+    next if raw_url.blank?
+    errors.add(:links, "Url not valid: #{key} - #{raw_url}") unless url_valid?(raw_url)
+  end
```

Comment:
Nice update. using links.each with key and raw_url improves clarity in error reporting. Skipping blank values is a good call too.